### ISSUE TITLE
Support for images NOT on docker hub?

### ISSUE THREAD

**Issue by dlasher** [jjlin/docker-image-extract#5]
Trying to extract an image hosted somewhere other than docker hub, and getting:

```
root@frigate:~# ./docker-image-extract ghcr.io/blakeblackshear/frigate:stable
Getting multi-arch manifest list...
No image digest found. Verify that the image, ref, and platform are valid.

root@frigate:~# ./docker-image-extract ghcr.io/blakeblackshear/frigate:stable-amd64
Getting multi-arch manifest list...
No image digest found. Verify that the image, ref, and platform are valid.
```

I suspect this is our problem:
```
# https://docs.docker.com/docker-hub/api/latest/#tag/repositories
manifest_list_url="https://hub.docker.com/v2/repositories/${image}/tags/${ref}"
```

Would be great to be able to target an image somewhere other than docker hub.

**Comment by creshal** [comment]
Yeah, it would be helpful if that was configurable. We would like to use our [Gitlab Dependency Proxy](https://docs.gitlab.com/ee/user/packages/dependency_proxy/) to cache images.

**Comment by jjlin** [comment]
I can probably make some changes for GHCR support at least, but out of curiosity, what are you guys using this script for? I mainly wrote this script for use cases where installing Docker wasn't feasible, but if it is, the real tools are going to be more robust.

**Comment by creshal** [comment]
The "real" tools are mostly more bloated. :) We need it for [vaultwarden, which recommends using docker-image-extract](https://github.com/dani-garcia/vaultwarden/wiki/Pre-built-binaries) for when running docker isn't feasible. It's a ridiculous workflow and I'd prefer for vaultwarden to just provide binaries directly, but until that happens, docker-image-extract is faster and easier to implement than adding another containerization layer to our CI.

**Comment by dlasher** [comment]
I wanted to unpack Frigate, so I could build it into an LXC container.  It used to be on docker hub (https://hub.docker.com/r/blakeblackshear/frigate) but then moved to https://ghcr.io/blakeblackshear/frigate:stable
